components/Navbar.vue

```diff
@@ -1,170 +1,240 @@
+            Groups
+          </li>
+        </nuxt-link>
+        <nuxt-link to="/csoc/resources">
+          <li class="list_item" @click="closeNavbar" @keydown="closeNavbar">
+            Resources
+          </li>
+        </nuxt-link>
+        <nuxt-link to="/csoc/timeline">
+          <li class="list_item" @click="closeNavbar" @keydown="closeNavbar">
+            Timeline
+          </li>
+        </nuxt-link>
+      </ul>
+    </div>
+  </div>
 </template>
 
-<script lang="ts">
-import {
-  mdiLaptopChromebook,
-  mdiBookshelf,
-  mdiMapClock,
-  mdiAccountGroup,
-  mdiChartLineVariant,
-  mdiGithub,
-  mdiCircleSlice3,
-  mdiHome,
-  mdiBlogger,
-} from '@mdi/js'
-
+<script>
 export default {
   data() {
     return {
-      sideNav: false,
-      links: [
-        { icon: mdiHome, title: 'Home', route: '/', hover: false },
-        { icon: mdiBlogger, title: 'Blog', route: '/blogs', hover: false },
-      ],
-      multiLink: [
-        {
-          icon: mdiLaptopChromebook,
-          title: 'CSOC',
-          route: '',
-          subroutes: [
-            {
-              icon: mdiBookshelf,
-              title: 'Resources',
-              route: '/csoc/resources',
-            },
-            {
-              icon: mdiMapClock,
-              title: 'Timeline',
-              route: '/csoc/timeline',
-            },
-          ],
-        },
-        {
-          icon: mdiAccountGroup,
-          title: 'Groups',
-          route: '',
-          subroutes: [
-            {
-              icon: mdiChartLineVariant,
-              title: 'CP',
-              route: '/group/cp',
-            },
-            {
-              icon: mdiGithub,
-              title: 'Dev',
-              route: '/group/dev',
-            },
-            {
-              icon: mdiCircleSlice3,
-              title: 'ML',
-              route: '/group/ml',
-            },
-          ],
-        },
-      ],
+      openTimeline: null,
+      isActive: false,
+      isOpen: false,
     }
   },
+  mounted() {
+    this.configNavbar()
+  },
+  methods: {
+    configNavbar() {
+      this.openTimeline = this.$gsap
+        .timeline({ paused: true })
+        .to('.navBg', {
+          duration: 0.8,
+          height: '100vh',
+          stagger: 0.3,
+        })
+        .to('#navTitle', { duration: 0.6, opacity: 1 })
+        .to('.list_item', { duration: 0.6, y: 0, stagger: 0.15, opacity: 1 }, 1)
+    },
+    async openNavbar() {
+      if (this.isActive) {
+        this.isActive = false
+        await this.openTimeline.reverse().then(function () {
+          this.isOpen = false
+        })
+      } else {
+        this.openTimeline.play()
+        this.isActive = true
+        this.isOpen = true
+      }
+    },
+    async closeNavbar() {
+      this.isActive = false
+      await this.openTimeline.reverse().then(function () {
+        this.isOpen = false
+      })
+    },
+  },
 }
 </script>
+
+<style scoped>
+/* BUTTON STYLING */
+.navbar-toggle {
```

Comment:
CSS to tailwind changes could have been made here also, but otherwise everything looks fine to me and the PR can be merged.
I have already made the change for the main website but you still can make the change since it will only help you keep CSS to minimal.